webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/ChunkUtils.scala

```diff
@@ -7,35 +7,35 @@ object ChunkUtils extends LazyLogging {
                           arrayChunkSize: Array[Int],
                           selectedShape: Array[Int],
                           selectedOffset: Array[Int]): List[Array[Int]] = {
-    val depth = arrayShape.length
-    val start = new Array[Int](depth)
-    val to = new Array[Int](depth)
+    val nDims = arrayShape.length
+    val start = new Array[Int](nDims)
+    val to = new Array[Int](nDims)
     var numChunks = 1
-    for (dim <- 0 until depth) {
-      val maxIdx = (arrayShape(dim) - 1) / arrayChunkSize(dim)
-      val startIdx = Math.min(maxIdx, selectedOffset(dim) / arrayChunkSize(dim))
-      var toIdx = (selectedOffset(dim) + selectedShape(dim) - 1) / arrayChunkSize(dim)
-      toIdx = Math.min(toIdx, maxIdx)
-      start(dim) = startIdx
-      to(dim) = toIdx
-      val numChunksForDim = toIdx - startIdx + 1
+    for (dim <- 0 until nDims) {
+      val largestPossibleIndex = (arrayShape(dim) - 1) / arrayChunkSize(dim)
+      val smallestPossibleIndex = 0
+      val startIndexRaw = selectedOffset(dim) / arrayChunkSize(dim)
+      val startIndexClamped = Math.max(smallestPossibleIndex, Math.min(largestPossibleIndex, startIndexRaw))
+      val toIndexRaw = (selectedOffset(dim) + selectedShape(dim) - 1) / arrayChunkSize(dim)
+      val toIndexClampedToBbox = Math.max(smallestPossibleIndex, Math.min(largestPossibleIndex, toIndexRaw))
+      val toIndexClamped = Math.max(startIndexClamped, toIndexClampedToBbox) // to index must be greater or equal to start index
+      start(dim) = startIndexClamped
+      to(dim) = toIndexClamped
+      val numChunksForDim = toIndexClamped - startIndexClamped + 1
       numChunks *= numChunksForDim
     }
-    if (numChunks < 0) {
-      logger.warn(
-        s"Failed to compute chunk indices. array shape ${arrayShape.toList}, chunkShape: ${arrayChunkSize.toList}, requested ${selectedShape.toList} at ${selectedOffset.toList}")
-    }
+
     val chunkIndices = new Array[Array[Int]](numChunks)
-    val currentIdx = start.clone
+    val currentIndex = start.clone
     for (i <- chunkIndices.indices) {
-      chunkIndices(i) = currentIdx.clone
-      var depthIdx = depth - 1
-      while ({ depthIdx >= 0 }) if (currentIdx(depthIdx) >= to(depthIdx)) {
-        currentIdx(depthIdx) = start(depthIdx)
-        depthIdx -= 1
+      chunkIndices(i) = currentIndex.clone
+      var dimIdx = nDims - 1
```

Comment:
```suggestion
      var dimIndex = nDims - 1
```
Is also changed above, should be changed here as well